### DISCUSSION TITLE
[WASM][Threading] JS Exception when using websockets.

### DISCUSSION THREAD

**Discussion by aquino-a** [unoplatform/uno#10799] [Q&A] [answered]
I recently upgraded a project to the latest UNO and .NET 7 in order to use threading in the WebAssembly implementation.
The project works fine with threading **disabled**.

### Websocket Issue Description:
- Logic is executed from a System.Threading.Timer.
- Is able to connect. ('Open' State)
- Javascript error when sending on the socket:
   'TypeError: Failed to execute 'decode' on 'TextDecoder': The provided ArrayBufferView value must not be shared.'
- If sent on the UIDispatcher, it doesn't have an error.
  In that case, the websocket is Closed Abnormally (1006) when receiving.

~~Is this an underlying issue or is websocket functionality all supposed to be run from the UI thread?~~

After looking further it looks like an issue on the dotnet side: https://github.com/dotnet/runtime/issues/74413

**Comment by jeromelaban** [comment, marked as answer]
This is indeed a .NET runtime issue. It will be fixed once fixed over there, then integrated into the Uno.Wasm.Bootstrap package.
